### ISSUE TITLE
Add Example for installing and using addons - RFC

### ISSUE THREAD

**Issue by joebo** [jsoftware/j-playground#71]
Now that addons are supported, we should have an example that highlights its use.

The example can highlight installing from github as well as a core addon once https://github.com/jsoftware/j-playground/issues/64#issuecomment-1137896376 is implemented.

What addons would we want to highlight?

My proposal:
1. The jreference addon could be useful (install 'github:AshyIsMe/jreference' )
2. The calculus addon would show off some J power (install 'github:jsoftware/math_calculus'), but I don't have any relevant examples to show it off yet 

Other ideas are welcome or someone taking a stab at creating the example

**Comment by bobtherriault** [comment]
If SVG is supported, I could try a version of jig that would allow users to see more information about the results that they create.

https://github.com/jsoftware/debug_jig

I would have to work with you on the nature of input. Currently, the user selects a line and the result is returned in a separate window in jqt using a short cut.

Cheers, bob

**Comment by cdburke** [comment]
Joe - great work, thanks. I made a start on an Addons script with your own example, but this could expand to a very nice set of examples.

**Comment by joebo** [comment]
> If SVG is supported, I could try a version of jig that would allow users to see more information about the results that they create. https://github.com/jsoftware/debug_jig I would have to work with you on the nature of input. Currently, the user selects a line and the result is returned in a separate window in jqt using a short cut. Cheers, bob

SVG should be supported as an output in the terminal. Here is an example.

```
CIRCLE =: 0 : 0
<svg height="100" width="100">
  <circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red" />
</svg> 
)
(2!:0) 'tcmhtml("<h1>fooooo</h1>")'
(2!:0) 'tcmhtml(jgetstr("CIRCLE"))'
```
[J Playground](https://jsoftware.github.io/j-playground/bin/html2/#code=CIRCLE%20%3D%3A%200%20%3A%200%0A%3Csvg%20height%3D%22100%22%20width%3D%22100%22%3E%0A%20%20%3Ccircle%20cx%3D%2250%22%20cy%3D%2250%22%20r%3D%2240%22%20stroke%3D%22black%22%20stroke-width%3D%223%22%20fill%3D%22red%22%20%2F%3E%0A%3C%2Fsvg%3E%20%0A%29%0A%282!%3A0%29%20'tcmhtml%28%22%3Ch1%3Efooooo%3C%2Fh1%3E%22%29'%0A%282!%3A0%29%20'tcmhtml%28jgetstr%28%22CIRCLE%22%29%29')

Or using the popup facility
```
POPUPHTML =: 0 : 0
<div>
Select a circle color to draw. The circle will draw on selection<br>
<select id="color" onchange="drawCircle()">
  <option>red</option>
  <option>blue</option>
</select>
</div>
<div id="circle">
</div>
<div style='text-align:right'><button id='popup-ok'>OK</button></div>
)

POPUPJS =: 0 : 0
  window.drawCircle = function() {
      var color = document.getElementById('color').value;
      circle.innerHTML = jdo1("getCircle '" + color + "'");
  }
)

NB. example of a j function returning SVG
NB. getCircle 'blue'
getCircle =: 3 : 0
'<svg height="100" width="100"><circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="', y, '" /></svg>'
)

NB. initialize JS
(2!:0) POPUPJS
(2!:0) 'popup(jgetstr("POPUPHTML"),500,"ok")'
```
[J Playground](https://jsoftware.github.io/j-playground/bin/html2/#code=POPUPHTML%20%3D%3A%200%20%3A%200%0A%3Cdiv%3E%0ASelect%20a%20circle%20color%20to%20draw.%20The%20circle%20will%20draw%20on%20selection%3Cbr%3E%0A%3Cselect%20id%3D%22color%22%20onchange%3D%22drawCircle%28%29%22%3E%0A%20%20%3Coption%3Ered%3C%2Foption%3E%0A%20%20%3Coption%3Eblue%3C%2Foption%3E%0A%3C%2Fselect%3E%0A%3C%2Fdiv%3E%0A%3Cdiv%20id%3D%22circle%22%3E%0A%3C%2Fdiv%3E%0A%3Cdiv%20style%3D'text-align%3Aright'%3E%3Cbutton%20id%3D'popup-ok'%3EOK%3C%2Fbutton%3E%3C%2Fdiv%3E%0A%29%0A%0APOPUPJS%20%3D%3A%200%20%3A%200%0A%20%20window.drawCircle%20%3D%20function%28%29%20%7B%0A%20%20%20%20%20%20var%20color%20%3D%20document.getElementById%28'color'%29.value%3B%0A%20%20%20%20%20%20circle.innerHTML%20%3D%20jdo1%28%22getCircle%20'%22%20%2B%20color%20%2B%20%22'%22%29%3B%0A%20%20%7D%0A%29%0A%0ANB.%20example%20of%20a%20j%20function%20returning%20SVG%0ANB.%20getCircle%20'blue'%0AgetCircle%20%3D%3A%203%20%3A%200%0A'%3Csvg%20height%3D%22100%22%20width%3D%22100%22%3E%3Ccircle%20cx%3D%2250%22%20cy%3D%2250%22%20r%3D%2240%22%20stroke%3D%22black%22%20stroke-width%3D%223%22%20fill%3D%22'%2C%20y%2C%20'%22%20%2F%3E%3C%2Fsvg%3E'%0A%29%0A%0ANB.%20initialize%20JS%0A%282!%3A0%29%20POPUPJS%0A%282!%3A0%29%20'popup%28jgetstr%28%22POPUPHTML%22%29%2C500%2C%22ok%22%29')

**Comment by joebo** [comment]
I also added this to the Examples menu. Feel free to suggest a better version of it!

**Comment by rdm** [comment]
When I run all lines on this UI example, I am seeing double in the session log -- assignment results echo.

I remember #53 touching on this issue. (Should #53 be opened? or should this be a new issue? or is this so trivial that its mere mention here is enough?)

**Comment by joebo** [comment]
> When I run all lines on this UI example, I am seeing double in the session log -- assignment results echo.

Good eye - I think this issue was never fixed for multi-line cases, which the UI example has many of. I've reopened #58 with a small reproducible example

**Comment by joebo** [comment]
> > When I run all lines on this UI example, I am seeing double in the session log -- assignment results echo.
> 
> Good eye - I think this issue was never fixed for multi-line cases, which the UI example has many of. I've reopened #58 with a small reproducible example

This is fixed now and #58 is closed again. Thanks again for the report
